app/src/main/java/com/num/controller/managers/DnsManager.java

```diff
@@ -0,0 +1,152 @@
+package com.num.controller.managers;
+
+import android.content.Context;
+import android.os.Handler;
+import android.util.Log;
+import android.util.Pair;
+
+import com.mobilyzer.MeasurementTask;
+//import com.mobilyzer.api.API;
+//import com.mobilyzer.exceptions.MeasurementError;
+import com.num.Constants;
+import com.num.controller.tasks.DnsLookupTask;
+import com.num.controller.tasks.LatencyTask;
+import com.num.controller.utils.PingUtil;
+import com.num.controller.utils.ServerUtil;
+import com.num.model.Address;
+import com.num.model.Dns;
+import com.num.model.Measurement;
+
+import java.util.ArrayList;
+//import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+//import java.util.Map;
+//import java.util.Objects;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import android.os.Message;
+
+/**
+ *
+ */
+public class DnsManager {
+
+    private static ThreadPoolExecutor dnsThreadPool;
+    private static BlockingQueue<Runnable> workQueue;
+    private Context context;
+    private List<Dns> dnses;
+    Measurement measurement;
+    private int count;
+    //private Handler handler;
+
+    /**
+     * Initialize a work queue and a thread pool.
+     * @param context
+     */
+    public DnsManager(Context context) {
+        workQueue = new LinkedBlockingQueue<>();
+        dnsThreadPool = new ThreadPoolExecutor(Constants.CORE_POOL_SIZE,
+                Constants.MAX_POOL_SIZE, Constants.KEEP_ALIVE_TIME, TimeUnit.SECONDS, workQueue);
+        //this.handler = handler;
+        this.context = context;
+        dnses = new ArrayList<>();
+
```

Comment:
yep, I eventually want to tune up these numbers once we have the dns list defined.
This keepalive is how much a thread will wait to be reused after completing a job. Once it times out the thread will be removed from the pool.
We are using a very basic approach for this dns task. If we want to measure things such as diplicated dns responses I believe we might want to go with a rich library such as dnsjava.org instead.